upload/src/addons/SV/ThreadReplyBanner/XF/Pub/Controller/Thread.php

```diff
@@ -5,6 +5,69 @@
 /*
  * Extends \XF\Pub\Controller\Thread
  */
+use XF\Mvc\ParameterBag;
+use XF\Mvc\Reply\View;
+use XF\Repository\Notice;
+
 class Thread extends XFCP_Thread
 {
+	public function actionReplyBannerHistory(ParameterBag $params) {
+		return $this->rerouteController(
+			'XF:EditHistory', 'index',
+			[
+				'content_type' => 'thread_banner',
+				'content_id'   => $params->get('thread_id')
+			]
+		);
+	}
+
+	public function actionIndex(ParameterBag $params) {
+		$reply = parent::actionIndex($params);
+
+		if ($reply instanceof View) {
+			if ($reply->getParam('thread')->getRelation('ThreadBanner')) {
```

Comment:
Instead of adding this as part of the response, extend the Thread entity to add `getThreadBanner()` helper method which returns this information. This way the template will pull the required information instead of having to push it into every template.